src/templates/partials/lighttpd.hbs

```diff
@@ -38,18 +19,40 @@ ssl.privkey = "/path/to/private_key"
 ssl.pemfile = "/path/to/signed_cert_followed_by_intermediates"
  {{#if (minver "1.0.2" form.opensslVersion)}}
   {{#if (minver "1.1.0" form.opensslVersion)}}
-ssl.openssl.ssl-conf-cmd = ("MinProtocol" => {{#if (includes "TLSv1" output.protocols)}}"TLSv1"{{else if (includes "TLSv1.1" output.protocols)}}"TLSv1.1"{{else if (includes "TLSv1.2" output.protocols)}}"TLSv1.2"{{else}}"TLSv1.3"{{/if}})
+   {{#if (includes "TLSv1" output.protocols)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1")
+   {{else if (includes "TLSv1.1" output.protocols)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.1")
+   {{else if (includes "TLSv1.2" output.protocols)}}
+    {{#if (minver "1.4.56" form.serverVersion)}}{{else}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")
+    {{/if}}
+   {{else}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.3")
+   {{/if}}
```

Comment:
@gstrauss It seems like this logic doesn't work. When I check this PR out and run it, selecting `Intermediate` and lighthttpd version `1.4.67` I get

```
...
ssl.pemfile = "/path/to/signed_cert_followed_by_intermediates"
ssl.openssl.ssl-conf-cmd += ("Options" => "-ServerPreference")
# TLS modules besides mod_openssl might name ciphers differently
...
```

And the `ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")` line is missing.

I'm unsure of the logic you intended.

The `Intermediate` configuration has a `output.protocols` list of `["TLSv1.2", "TLSv1.3"]`

* What `MinProtocol` should lighthttpd have for `Intermediate` on version `1.4.67`?

Also if you change the `{{#if (minver "1.4.56" form.serverVersion)}}{{else}}` line to use `unless` instead of `if` with an empty `else` it will fix the formatting issue that causes the line to be preceded by 4 spaces causing it to look like

```
    ssl.openssl.ssl-conf-cmd += ("Options" => "-ServerPreference")
```

Here's what it would look like with `unless`

```
    {{#unless (minver "1.4.56" form.serverVersion)}}
ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")
    {{/unless}}
```

Reply:
`"MinProtocol" => "TLSv1.2"` is the default in lighttpd TLS modules since lighttpd 1.4.56.
It is preferred that people use lighttpd's secure defaults -- which are periodically reviewed, announced, and updated -- rather than specifying something which may then be cargo-culted for years and never revisited in user configurations.

I'll make the suggested `unless` change shortly and will push an updated branch rebased on master.

Reply:
requested changes have been made

Reply:
FYI, the commit message for a56011b7 in this PR explains:
```
lighttpd updates for lighttpd 1.4.68

lighttpd 1.4.56 and later default to MinProtocol TLSv1.2

lighttpd 1.4.68 and later default to a strict set of PFS ciphers and
to -ServerPreference, since a strict set of PFS ciphers is the default.

Simplify intermediate and modern configs for lighttpd 1.4.68 and later.
lighttpd defaults will incrementally be made more secure in the future
and using lighttpd secure defaults (without explicitly hard-coding the
defaults at a point in time) will allow users to get more secure
defaults along with lighttpd upgrades, instead of accidentally
continuing to use explicitly set older, less-secure config settings.
```

Reply:
Besides rebasing this PR onto tip of current master branch, the only substantive change in the push below is your requested change to use `unless` in `src/templates/partials/lighttpd.hbs`

Reply:
Not sure if this should include 308 redirects, that ought to be addressed separately #117 — there were discussions in #137 regarding browser support and selective addition to only modern profile etc.; there was also impact to .json specs so that would need to get versioned etc. …

Reply:
@janbrasna have you looked at https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308 recently?  All modern, supported browser clients support 308.  If you have a specific concern, please be more specific.  "Fear, Uncertainty, and Doubt" absent any **real data** are not strong arguments or productive contributions to a conversation.

[RFC7538](https://www.rfc-editor.org/rfc/rfc7538) "The Hypertext Transfer Protocol Status Code 308 (Permanent Redirect)" was published in 2015, **almost 9 years ago**.

Reply:
@gstrauss Thanks for the `unless` fix to the indentation.

> `"MinProtocol" => "TLSv1.2"` is the default in lighttpd TLS modules since lighttpd 1.4.56.

Got it, what I meant is that it looks like the logic you wrote is intended to output the line

```
ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")
```

However it doesn't.

Even with the default you mention, will the subsequent line

```
    ssl.openssl.ssl-conf-cmd += ("Options" => "-ServerPreference")
```

work given that it's using a `+=` which I imagine concatenates onto an existing string? I ask because in the code's current state there is no existing string to add to.

Here's the output when I run this PR's code and choose lighthttp 1.4.67

![image](https://github.com/mozilla/ssl-config-generator/assets/1134034/f773b32e-d949-4659-881e-f4f547b861d0)

Ya, I'd been intending to try to work on the HTTP 308 topic distinct from this PR, but don't worry about that, I can fix this up. Once we have the logic for the `MinProtocol` fixed I'll fix things up and add more comments to #117

Reply:
> Even with the default you mention, will the subsequent line
>
>    `ssl.openssl.ssl-conf-cmd += ("Options" => "-ServerPreference")`
>
> work given that it's using a += which I imagine concatenates onto an existing string? I ask because in the code's
> current state there is no existing string to add to.

@gene1wood: Yes, it will work as intended.

Reply:
Ok cool, ya appending to empty lists like this was added in [1.4.17](https://redmine.lighttpd.net/versions/8) as you say.

Reply:
@gstrauss No need to worry about me, I work with MDN daily. If pointing to previous maintainers' decisions (as a reason why more related HTTP 308 issues/PRs are still hanging around unresolved) feels like a FUD then I'm sorry, but I felt co-authors' decisions were more relevant, than e. g. my recollections of why this hasn't moved forward. That said… I just tried to explain that separating the 308 feature into its own might help expedite the proposed changes.

(There's no consensus on 308s and once decided, that should be projected into all server templates, behind their appropriate version conditionals i. e. ~ sure I'd love to see them given the POST handling, but historically there was the question if they belong to `intermediate` where IE11@W7 is buggy and `old` where most of the UAs claimed to be compatible can't redirect it. That's why I wanted to move this discussion to its own issue, and iron out the details there.)

Reply:
I have removed the 308 changes from this PR.  First, the lighttpd 1.4.75 was released over 7 months ago and does something better by default.  Second, I'm willing to continue the 308 discussion knowing that I can push for a decision/resolution (one way or another) in a reasonable timeframe, e.g. within a couple months.